### PR TITLE
Add user registration and authentication

### PR DESCRIPTION
**Pull request by AleksandraCyp**
Install new dependencies and add JWT_PRIVATE_KEY to development.env

### CODE REVIEW

File: src/entities/Auth/Auth.validation.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import Joi, { ValidationResult } from "joi";
+import AuthReq from './Auth.interface';
+
+const validateAuthReq = (authReq: AuthReq): ValidationResult => {
+    const schema = Joi.object({
+        // eslint-disable-next-line max-len
+        email: Joi.string().min(5).max(255).required().email().pattern(/[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?/),
```
Comment (lukaszdutka) [inline src/entities/Auth/Auth.validation.ts]:
```suggestion
        email: Joi.string()
        .min(5)
        .max(255)
        .required()
        .email()
        .pattern(/[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?/), 
```

Comment (lukaszdutka) [reply src/entities/Auth/Auth.validation.ts]: Isn't `.email()` validation enough to ensure that this is email? Why adding `.pattern(...)`with regex?

File: src/entities/Auth/Auth.validation.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import Joi, { ValidationResult } from "joi";
+import AuthReq from './Auth.interface';
+
+const validateAuthReq = (authReq: AuthReq): ValidationResult => {
+    const schema = Joi.object({
+        // eslint-disable-next-line max-len
+        email: Joi.string().min(5).max(255).required().email().pattern(/[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?/), 
+        // eslint-disable-next-line max-len
+        password: Joi.string().min(5).max(255).required()
```
Comment (lukaszdutka) [inline src/entities/Auth/Auth.validation.ts:9]:
```suggestion
        password: Joi.string()
        .min(5)
        .max(255)
        .required()
```

File: src/routes/Auth/Auth.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+import jwt from 'jsonwebtoken';
+import userModel from '../../entities/User/User.schema';
+import validateAuthReq from '../../entities/Auth/Auth.validation';
+
+const router = Router();
+const { BAD_REQUEST } = StatusCodes;
+const User = userModel;
+
+// authenticate users
+// eslint-disable-next-line @typescript-eslint/no-misused-promises
```
Comment (lukaszdutka) [inline src/routes/Auth/Auth.ts]: this is not a good way to handle eslint warnings. 
It should be used as a hint to find where something is not right. 
I am 95% sure you can hover over the warnings and see some valuable feedback from linter, why the code gives warnings

**Review by memeraki — APPROVED** [review]
works 👌 
and user doesn't have to be 'any' anymore 😆

**Review by daria305 — APPROVED** [review]
All works fine, I am merging it

**Pull request merged**